recipes/Lyra/all/conanfile.py

```diff
@@ -0,0 +1,28 @@
+from conans import ConanFile, tools
+import os.path
+
+
+class LyraConan(ConanFile):
+    name = "lyra"
+    homepage = "https://bfgroup.github.io/Lyra/"
+    description = "A simple to use, composing, header only, command line arguments parser for C++ 11 and beyond."
+    topics = ("conan", "cli", "c++11")
+    author = "Build Frameworks Group"
+
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "os", "compiler", "arch", "build_type"
```

Comment:
```suggestion
```
It's not requires, since is a header-only recipe.

Reply:
Since your recipe doesn't use settings anywhere it doesn't need to define them. 
And since you don't need them - you also don't need the `package_id` function.

Reply:
@grafikrobot Conan doesn't have a built-in way to indicate the type of the library (or its components). The `header_only` function's function is not to indicate but to adjust the hash.
I mean, you can still use it, but it's missing the point.
And yes, to have a mechanism to indicate the type of a library's component would be great.